tests/userSocialMedia.test.ts

```diff
@@ -0,0 +1,121 @@
+  await DatabaseConnection.close();
+});
+
+describe('social media URL submission', () => {
+  test('properly persists on successful submission', async () => {
+    const conn = await DatabaseConnection.get();
+    let member = UserFactory.fake();
+    const userSocialMedia = UserSocialMediaFactory.fake({ user: member, type: SocialMediaType.FACEBOOK });
+
+    await new PortalState()
+      .createUsers(member)
+      .write();
+
+    const userController = ControllerFactory.user(conn);
+    await userController.insertSocialMediaForUser({ socialMedia: userSocialMedia }, member);
+    member = await conn.manager.findOne(UserModel, { uuid: member.uuid }, { relations: ['userSocialMedia'] });
+
+    expect(member.userSocialMedia).toHaveLength(1);
+    expect(member.userSocialMedia[0]).toEqual({
+      url: userSocialMedia.url,
+      type: userSocialMedia.type,
+      uuid: userSocialMedia.uuid,
+    });
+  });
+
+  test('is invalidated when submitting social media URL of the same type', async () => {
+    const conn = await DatabaseConnection.get();
+    const member = UserFactory.fake();
+    const userSocialMedia = UserSocialMediaFactory.fake({ user: member, type: SocialMediaType.FACEBOOK });
+
+    await new PortalState()
+      .createUsers(member)
+      .createUserSocialMedia(userSocialMedia)
+      .write();
+
+    const userController = ControllerFactory.user(conn);
+
+    const userSocialMediaWithSameType = UserSocialMediaFactory.fake({ type: SocialMediaType.FACEBOOK });
+    const errorMessage = 'Social media URL of this type has already been created for this user';
+    await expect(userController.insertSocialMediaForUser({ socialMedia: userSocialMediaWithSameType }, member))
+      .rejects.toThrow(errorMessage);
+  });
+});
+
+describe('social media URL update', () => {
+  test('is invalidated when updating social media URL of another user', async () => {
+    const conn = await DatabaseConnection.get();
+    let member = UserFactory.fake();
+    const unauthorizedMember = UserFactory.fake();
+    const userSocialMedia = UserSocialMediaFactory.fake({ user: member, type: SocialMediaType.FACEBOOK });
+
+    await new PortalState()
+      .createUsers(member, unauthorizedMember)
+      .createUserSocialMedia(userSocialMedia)
+      .write();
+
+    const userController = ControllerFactory.user(conn);
+    member = await conn.manager.findOne(UserModel, { uuid: member.uuid }, { relations: ['userSocialMedia'] });
+
+    const errorMessage = 'User cannot update a social media URL of another user';
+    const uuidParams = { uuid: member.userSocialMedia[0].uuid };
+    const socialMediaParams = { socialMedia: { url: faker.internet.url() } };
+    await expect(userController.updateSocialMediaForUser(uuidParams, socialMediaParams, unauthorizedMember))
+      .rejects.toThrow(errorMessage);
+  });
+
+  test('is invalidated when invalid uuid is passed in', async () => {
+    const conn = await DatabaseConnection.get();
```

Comment:
nit: 'invalid uuid' here can mean a few different things (e.g. an improperly formatted one, a uuid v2 vs. a v4, etc.) so let's be more specific here and with the variable naming on what exactly we are trying to test. It seems like we are testing a uuid that is valid but that is not found, so we can rename this test to 'is invalidated when the entity for a uuid is not found', and rename `uuidParams` to something like `missingEntityUuid`